Sources/SwiftParser/Availability.swift

```diff
@@ -262,48 +262,97 @@ extension Parser {
   ///     version-list -> version-tuple-element version-list?
   ///     version-tuple-element -> '.' interger-literal
   mutating func parseVersionTuple(maxComponentCount: Int) -> RawVersionTupleSyntax {
-    if self.at(.floatingLiteral),
-      let periodIndex = self.currentToken.tokenText.firstIndex(of: UInt8(ascii: ".")),
-      self.currentToken.tokenText[0..<periodIndex].allSatisfy({ Unicode.Scalar($0).isDigit })
-    {
-      // The lexer generates a float literal '1.2' for the major and minor version.
-      // Split it into two integers if possible
-      let major = self.consumePrefix(SyntaxText(rebasing: self.currentToken.tokenText[0..<periodIndex]), as: .integerLiteral)
+    if self.at(.floatingLiteral) {
 
-      var components: [RawVersionComponentSyntax] = []
-      var trailingComponents: [RawVersionComponentSyntax] = []
+      if let periodIndex = self.currentToken.tokenText.firstIndex(of: UInt8(ascii: ".")),
+        self.currentToken.tokenText[0..<periodIndex].allSatisfy({ Unicode.Scalar($0).isDigit })
+      {
+        // The lexer generates a float literal '1.2' for the major and minor version.
+        // Split it into two integers if possible
+        let major = self.consumePrefix(SyntaxText(rebasing: self.currentToken.tokenText[0..<periodIndex]), as: .integerLiteral)
 
-      for i in 1... {
-        guard let period = self.consume(if: .period) else {
-          break
-        }
-        let version = self.expectDecimalIntegerWithoutRecovery()
+        var components: [RawVersionComponentSyntax] = []
+        var trailingComponents: [RawVersionComponentSyntax] = []
 
-        let versionComponent = RawVersionComponentSyntax(period: period, number: version, arena: self.arena)
+        for i in 1... {
+          guard let period = self.consume(if: .period) else {
+            break
+          }
+          let version = self.expectDecimalIntegerWithoutRecovery()
 
-        if i < maxComponentCount {
-          components.append(versionComponent)
-        } else {
-          trailingComponents.append(versionComponent)
-        }
-      }
+          let versionComponent = RawVersionComponentSyntax(period: period, number: version, arena: self.arena)
 
-      var unexpectedTrailingComponents: RawUnexpectedNodesSyntax?
+          if i < maxComponentCount {
+            components.append(versionComponent)
+          } else {
+            trailingComponents.append(versionComponent)
+          }
 
-      if !trailingComponents.isEmpty {
-        unexpectedTrailingComponents = RawUnexpectedNodesSyntax(elements: trailingComponents.compactMap { $0.as(RawSyntax.self) }, arena: self.arena)
-      }
+          if versionComponent.hasError {
```

Comment:
If the versionComponent has an error, the loop will be stopped, and both the major and the parsed components up to that point, as well as all tokens before the comma or right parenthesis, will be treated as unexpectedToken.

Reply:
I believe this incorrectly marks the entire version as invalid if you have eg. `1.2.0xff`. Ideally we should only be complaining about the `0xff` here.